### PR TITLE
Sept 23 upstream 4 (dc1f086)

### PR DESCRIPTION
**Pull request by illusional**
Depends on: https://github.com/populationgenomics/hail/pull/304

### CODE REVIEW

File: hail/python/hailtop/batch/backend.py [hunk 1]
```diff
@@ -803,19 +793,23 @@ async def compile_job(job):
...
         if verbose:
-            print(f'Batch consists of {n_jobs_submitted} jobs, submitted in {round(time.time() - submit_batch_start, 3)} seconds:')
+            print(f'Submitted batch {batch_id} with {n_jobs_submitted} jobs in {round(time.time() - submit_batch_start, 3)} seconds:')
             for jid, cmd in jobs_to_command.items():
                 print(f'{jid}: {cmd}')
             print('')
+
+        deploy_config = get_deploy_config()
+        url = deploy_config.external_url('batch', f'/batches/{batch_id}')
+
```
Comment (jmarshall) [inline hail/python/hailtop/batch/backend.py:803]: I think these lines are a merge accident.

File: batch/batch/batch.py [hunk 2]
```diff
@@ -56,7 +64,8 @@ def _time_msecs_str(t):
         'time_completed': time_completed,
         'duration': duration,
         'msec_mcpu': record['msec_mcpu'],
-        'cost': coalesce(record.get('cost'), 0),
+        'cost': coalesce(record['cost'], 0),
```
Comment (jmarshall) [inline batch/batch/batch.py:67]: This accidentally(?) dropped your change to this line from abd4476942:
```suggestion
        'cost': coalesce(record.get('cost'), 0),
```

File: ci/ci/github.py [hunk 3]
```diff
@@ -1031,15 +1078,15 @@ async def deploy(self, db: Database, batch_client, steps, excluded_steps=()):
                 }
             )
             config.build(deploy_batch, self, scope='dev')
-            deploy_batch = await deploy_batch.submit()
+            await deploy_batch.submit()
```
Comment (jmarshall) [inline ci/ci/github.py:1081]: There's another instance on line 979 that looks like it also needs `deploy_batch =` removed from it.

Comment (jmarshall) [reply ci/ci/github.py:1081]: I've got no real understanding of what's going on with this code, of course 😄

**Review by jmarshall — APPROVED** [review]
I've commented on all the potential problems I saw in this one.

There's also the secrets in _build.yaml_ where we seem to have removed some previously and some refactoring/renaming has gone on upstream. That seems like a suck it and see scenario, and you know better than I do what secrets we've got set up.

File: hail/python/hailtop/batch/backend.py (hunk 1 above)
Comment (jmarshall) [reply hail/python/hailtop/batch/backend.py:803]: Bumping this (minor) comment.

Comment (illusional) [reply hail/python/hailtop/batch/backend.py]: Thanks @jmarshall, have resolved by removing these lines

**Pull request merged**